File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        ans = -1
        left, right = 0, len(heights) - 1
        while left < right:
            area = (right - left) * min(heights[left], heights[right])
            ans = max(ans, area)
            if heights[left] <= heights[right]:
                left += 1
            else:
                right -= 1
        return ans

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int goodNodes(TreeNode root) {
        if (root == null) {
            return 0;
        }
        return dfs(root, 0);
    }

    private int dfs(TreeNode node, int currMax) {
        int sum = 0;
        if (node.val >= currMax) {
            sum += 1;
        }

        if (node.right != null) {
            sum += dfs(node.right, Math.max(node.val, currMax));
        }

        if(node.left != null) {
            sum += dfs(node.left, Math.max(node.val, currMax));
        }
        return sum;
    }
}
